src/EditorFeatures/VisualBasicTest/MakeTypeAbstract/MakeTypeAbstractTests.vb

```diff
@@ -22,7 +22,7 @@ Public Class [|Foo|]
     Public MustOverride Sub M()
 End Class",
 "
-Public MustOverride Class Foo
+Public MustInherit Class Foo
```

Comment:
given this example, i guess my surprise is that we are passing in hte class-statement, and not hte class-block for this feature.

Reply:
@CyrusNajmabadi, yep, it passes the class statement.

Reply:
i guess my fix would be at that level :) There is also an ISymbolDeclarationService that helps ensure that we work with the right node (if this is a feature that uses symbols).

Reply:
@CyrusNajmabadi The fix is mainly for the public `SyntaxGenerator.WithModifiers` API

Reply:
i guess my point is this: i think it's hte feature working improperly.  i would update the feature.  if we update the generator, we need to update it to support all statements for blocks, and it was intentional that we were not conflating these at that level.

Reply:
@CyrusNajmabadi For the purpose of `WithModifiers`, supporting other statements won't have effect. For `SyntaxGenerator` in general, I'm not sure what other parts are affected by but supporting blocks only and not statements?

Reply:
i'm not sure i get your question.  it feels very strange to just special case class statements here (what about structs/interfaces/modules)?

Reply:
@CyrusNajmabadi In this specific code path. They have no effect. The declaration kind is passed to `GetModifierList`:

https://github.com/dotnet/roslyn/blob/ad0674f544bfd0e5e6110f761ff8cae80a028544/src/Workspaces/VisualBasic/Portable/CodeGeneration/VisualBasicSyntaxGenerator.vb#L2490